### PR TITLE
BP-40: Add items

### PR DESCRIPTION
**Pull request by MrSachin7**

### CODE REVIEW

**Comment by MrSachin7** [comment]
Do not hit merge. Let the creator hit it.

File: src/Domain/item/ItemDomain.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+public class ItemDomain : IItemDomain {
+
+    private readonly IItemDao _itemDao;
+
+    public ItemDomain(IItemDao itemDao) {
+        _itemDao = itemDao;
+    }
+
+    public async Task<ItemDto> CreateItem(ItemDto itemDto) {
+        itemDto = new ItemDto(itemDto.Name.Trim());
```
Comment (himal28924) [inline src/Domain/item/ItemDomain.cs:17]: I guess trim only remove white space , so we might also need to check with : lower case and uppercase

Comment (MrSachin7) [reply src/Domain/item/ItemDomain.cs:17]: The use case document says nothing about lowercase or uppercase.  But then again we can change it. But what exactly needs to be checked with lowercase or uppercase ? Is it we only allow item names on lowercase ?

File: src/Data/dao/ItemDao.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+public class ItemDao : IItemDao {
+    private readonly DatabaseContext _databaseContext;
+
+    public ItemDao(DatabaseContext databaseContext) {
+        _databaseContext = databaseContext;
+    }
+
+    public async Task<ItemDto?> GetItemByName(string itemDtoName) {
+        ItemEntity? itemEntity = await _databaseContext.Items!.FindAsync(itemDtoName);
```
Comment (himal28924) [inline src/Data/dao/ItemDao.cs]: Check for lower case and uppercase

File: src/Domain/item/ItemDomain.cs (hunk 1 above)
Comment (MrSachin7) [reply src/Domain/item/ItemDomain.cs:17]: Now, the getItemByName checks for name ignoring the case. This should fix the issue we discussed about.

**Review by himal28924 — APPROVED** [review]
You have fixed it so merge it ASAP . :) 
Thank you for your time

**Pull request merged**